src/main/java/to/wetransform/hale/transformer/Transformer.java

```diff
@@ -1,65 +1,238 @@
+        // init working directory
+        // Create a temporary directory
+        Path tempDirectory = null;
+        try {
+            tempDirectory = Files.createTempDirectory("hale-transformer");
+            File workingFile = null;
 
-        ExecContext context = new ExecContext();
+            if (tempDirectory != null) {
+                workingFile = tempDirectory.toFile();
+            } else {
+                workingFile = run.createTempDir();
+            }
 
-        // URI projectUri = ....
-        // context.setProject(projectUri);
-        // Project project = loadProject(projectUri);
+            LOG.info("Using working directory " + tempDirectory.toString());
+            File reportFile = null;
+            try {
+                // setup report file
+                reportFile =
+                        Files.createTempFile(tempDirectory, "reports", ".log").toFile();
+                reportFile.delete();
+                reportFile.createNewFile();
+            } catch (Throwable t) {
+                LOG.error("Error in creating report log file.");
+                return;
+            }
 
-        // context.setSources(...)
-        // context.setSourceProviderIds(...)
-        // context.setSourcesSettings(...)
+            try {
+                // setup log file
+                transformationLogFile = Files.createTempFile(tempDirectory, "transformation", ".log")
+                        .toFile();
+                transformationLogFile.delete();
+                transformationLogFile.createNewFile();
 
-        // Value sourceCrs = null;
-        // TODO determine source CRS
+                // fork streams
+                FileOutputStream transformationLogOut = new FileOutputStream(transformationLogFile, true);
+                OutputStream tOut = new TeeOutputStream(sysOut, transformationLogOut);
+                System.setOut(new PrintStream(tOut));
+            } catch (Throwable t) {
+                LOG.error("Error in creating log file.");
+                return;
+            }
 
-        // TargetConfig targetConfig = configureTarget(project, sourceCrs);
+            LOG.info("Startup...");
+
+            long heapMaxSize = Runtime.getRuntime().maxMemory();
+            LOG.info("Maximum heap size configured as " + IOUtils.humanReadableByteCount(heapMaxSize, false));
+
+            // initialize
+            Init.init();
+
+            Version version = HalePlatform.getCoreVersion();
+            LOG.info(MessageFormat.format("Launching hale-transformer {0}...", version.toString()));
+
+            ExecContext context = new ExecContext();
+
+            // Set up project URI
+            URI projectUri = new URI(projectURL);
+            context.setProject(projectUri);
+
+            // Load project
+            Project project = loadProject(projectUri);
+
+            SourceConfig sourceConfig = new SourceConfig(
```

Comment:
@florianesser any thoughts on how to set up the `settings `for the `sourceConfig`. This is the reason why the `ExecTransformation` fails.
This is the line failing source.loadConfiguration(context.getSourcesSettings().get(index));` in the `ExecTransformation` line 312.